clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1241,6 +1326,8 @@ public void handleResponse(AbstractResponse response) {
                     log.debug("Did not attempt to add partition {} to transaction because other partitions in the " +
                             "batch had errors.", topicPartition);
                     hasPartitionErrors = true;
+                } else if (error == Errors.UNKNOWN_PRODUCER_ID || error == Errors.INVALID_PRODUCER_ID_MAPPING) {
```

Comment:
Why is `INVALID_PRODUCER_ID_MAPPING` no longer fatal? Does that not indicate we have likely been fenced?

Reply:
According to the KIP, it could mean that the producer ID had expired. On the server side, we'll generate a new ID and return it if there's no existing producer ID for the given transactional ID, or we'll fence the producer if a new producer ID already exists

Reply:
Ok, that makes sense. I had forgotten about it. If you can find a good place, I think this would be good to document.